src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/ActivityHelper.cs

```diff
@@ -63,25 +65,29 @@ public static bool HasStarted(HttpContext context, out Activity? aspNetActivity)
     {
         PropagationContext propagationContext = textMapPropagator.Extract(default, context.Request, HttpRequestHeaderValuesGetter);
 
-        KeyValuePair<string, object?>[]? tags;
-        if (context.Request?.Unvalidated?.Path is string path)
-        {
-            tags = cachedTagsStorage ??= new KeyValuePair<string, object?>[1];
+        string? path = context.Request?.Unvalidated?.Path;
+        string? originalHttpMethod = context.Request?.HttpMethod;
 
-            tags[0] = new KeyValuePair<string, object?>("url.path", path);
-        }
-        else
+        string normalizedHttpMethod = RequestDataHelper.GetNormalizedHttpMethod(originalHttpMethod ?? string.Empty);
+
+        List<KeyValuePair<string, object?>> tags = cachedTagsStorage ??= new List<KeyValuePair<string, object?>>(3); // use max capacity
+
+        if (path is not null)
         {
-            tags = null;
+            tags.Add(new KeyValuePair<string, object?>(SemanticConventions.AttributeUrlPath, path));
         }
 
-        Activity? activity = AspNetSource.StartActivity(TelemetryHttpModule.AspNetActivityName, ActivityKind.Server, propagationContext.ActivityContext, tags);
+        tags.Add(new KeyValuePair<string, object?>(SemanticConventions.AttributeHttpRequestMethod, normalizedHttpMethod));
```

Comment:
* We can probably remove this also being done in AspNet because sampler tags are automatically added to the span IIRC: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/32c4e943405acf811120b9723aabd3740f63b8ae/src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs#L77

* Consider refactoring this into `RequestDataHelper` something like this...

```csharp
RequestDataHelper.SetHttpMethodTag(tags, originalHttpMethod: context.Request.HttpMethod);

class RequestDataHelper
{
    public void SetHttpMethodTag(List<KeyValuePair<string, object?>> tags, string originalHttpMethod)
    {
        var normalizedHttpMethod = this.GetNormalizedHttpMethod(originalHttpMethod);
        tags.Add(new(SemanticConventions.AttributeHttpRequestMethod, normalizedHttpMethod));

        if (originalHttpMethod != normalizedHttpMethod)
        {
            tags.Add(new(SemanticConventions.AttributeHttpRequestMethodOriginal, originalHttpMethod));
        }
    }
}
```

Reason being it seems like all of this logic is already encapsulated inside `RequestDataHelper`.